std/process.d

```diff
@@ -1034,54 +1034,92 @@ private Pid spawnProcessPosix(scope const(char[])[] args,
 
             if (!(config.flags & Config.Flags.inheritFDs))
             {
-                // NOTE: malloc() and getrlimit() are not on the POSIX async
-                // signal safe functions list, but practically this should
-                // not be a problem. Java VM and CPython also use malloc()
-                // in its own implementation via opendir().
-                import core.stdc.stdlib : malloc;
-                import core.sys.posix.poll : pollfd, poll, POLLNVAL;
-                import core.sys.posix.sys.resource : rlimit, getrlimit, RLIMIT_NOFILE;
-
-                // Get the maximum number of file descriptors that could be open.
-                rlimit r;
-                if (getrlimit(RLIMIT_NOFILE, &r) != 0)
-                {
-                    abortOnError(forkPipeOut, InternalError.getrlimit, .errno);
-                }
-                immutable maxDescriptors = cast(int) r.rlim_cur;
+                import core.sys.posix.dirent : dirent, opendir, readdir, closedir, DIR;
+                import core.sys.posix.unistd : close;
+                import core.sys.posix.stdlib : atoi, malloc, free;
 
-                // The above, less stdin, stdout, and stderr
-                immutable maxToClose = maxDescriptors - 3;
+                // Missing druntime declaration
+                pragma(mangle, "dirfd")
+                extern(C) nothrow @nogc int dirfd(DIR* dir);
```

Comment:
Documentation on opengroup says 

> This interface was introduced because the Base Definitions volume of POSIX.1-2017 does not make public the DIR data structure. 

Unless the year is wrong it looks like the function might be too new to be considered.

Reply:
See the discussion above :)

Reply:
> See the discussion above :)

I've just seen it now. I'm firmly in the not sure camp, as I'll be the one who'll get stick for undefined reference issues on Solaris 11 and Darwin 12.

Reply:
Do we have any way of quickly checking?

Being able to run a command / try to compile a simple C program on a variety of OSes would be a really useful tool to have...

Reply:
The first version I put here indeed used a linked list of fds. I can restore it, if you want. 

About Solaris 11: https://stackoverflow.com/a/28025462

About Darwin: this page said those function were added on 4.2BSD. Isn't Darwin bsd based?